spec/integration/post_index_spec.rb

```diff
@@ -16,33 +16,44 @@
                          comments_counter: 0, author_id: @user1.id)
     @post5 = Post.create(title: 'Automobile engineering', text: 'Reading', likes_counter: 0,
                          comments_counter: 0, author_id: @user2.id)
-    Comment.create(text: 'This is great', post_id: @post1.id, author_id: @user2.id)
+    @comment1 = Comment.create(text: 'This is great', post_id: @post1.id, author_id: @user2.id)
+    visit user_path(@user1.id)
   end
 
   context 'Load index page' do
-    it 'Should see the posts title' do
-      visit posts_path(@user1.id)
-      expect(page).to have_content('This is a post')
+    it 'Should see the profile picture' do
+      expect(page).to have_selector('img')
     end
 
     it 'Should see the posts author' do
-      visit posts_path(@user1.id)
       expect(page).to have_content('Aisha')
     end
 
-    it 'Should see the posts comments counter' do
-      visit posts_path(@user1.id)
-      expect(page).to have_content('Comments(0)')
+    it 'Should see the number of posts' do
+      expect(page).to have_content('Number of posts: 4')
+    end
+
+    it 'Should see the posts title' do
+      expect(page).to have_content('This is a post')
+    end
+
+    it 'Should see the number of comments' do
+      expect(page).to have_content("Comments: #{@post1.comments_counter}")
+    end
+
+    it 'Should see the number of likes' do
+      expect(page).to have_content("Likes: #{@post1.likes_counter}")
     end
 
-    it 'Should see the posts likes counter' do
-      visit posts_path(@user1.id)
-      expect(page).to have_content('Likes(0)')
+    it 'Should see some post body and first comment' do
+      click_link 'This is a post'
+      expect(page).to have_content('Accounting made easy')
+      expect(page).to have_content('Moses:This is great')
     end
```

Comment:
- [ ] It would be very great if you separate both `expect` in different, it blocks because if you don't do it then there is no point in using it blocks if you wanna have all the `expects` in the same block :thinking: Also, if you in future any of your test fails then you won't be able to find out which test is failing because some of your expects would be sharing the same it blocks :fearful: Kindly do that as shown below :arrow_down:

```ruby
    it 'Should see some post body' do
      expect(page).to have_content('Accounting made easy')
    end

    it 'Should see the first comment on the post' do
      click_link 'This is a post'
      expect(page).to have_content('Moses:This is great')
    end
```

### Additional but Important Information :memo:

After making the above change one of your tests will fail and the test is about the body because you don't have the body of the post in your user_post_index page :fearful: It would be very great if you add the body of your post in that view so your test will pass :green_circle: Also, it is a requirement in the wireframe which has shown to you in the project requirements and you can see below :arrow_down:

![image](https://user-images.githubusercontent.com/93039359/226809745-b6a75388-cf99-4f8f-873f-09ced469d4ec.png)


:red_circle: **Questions**

1. There is no requirement of this project to change our views then why are you asking me to change my views?
2. I can test it by expecting the body on my user_post_show page then I don't need to change my user_post_index view :question:

:green_circle: **Answers**

1. You are right that there is no requirement in this project to change your views but as far as tests are concerned. Your test should pass to make them pass legally, you will need to change your view :blush:
2. Yeah you can expect it in the user_post_show page but you are writing tests in your user_post_index page so it would be nice if your tests would be related to user_index_page :+1: